collect_app/build.gradle

```diff
@@ -177,13 +173,6 @@ android {
         pickFirst 'draftv4/schema'
         pickFirst 'draftv3/schema'
 
-        // To ensure that ODK Collect is installable on all devices, we don't use
-        // abiFilters to exclude any ABIs; but to keep the APK slim, we include
-        // the Mapbox native library only for 32-bit and 64-bit ARM devices and
-        // omit it for all X86 devices.
-        exclude 'lib/x86/libmapbox-gl.so'
-        exclude 'lib/x86_64/libmapbox-gl.so'
```

Comment:
What crash are you seeing? I think we'll want to do as much as can to avoid including unused `x86` libs.

Reply:
Agreed with @seadowg that more details are needed here. What's the binary size without the filtering?

Reply:
The error is:
```
Caused by: java.lang.UnsatisfiedLinkError: dalvik.system.PathClassLoader[DexPathList[[zip file "/data/app/org.odk.collect.android-fpP4T_tgz8x_GU7w9kKqvA==/base.apk"],nativeLibraryDirectories=[/data/app/org.odk.collect.android-fpP4T_tgz8x_GU7w9kKqvA==/lib/x86_64, /data/app/org.odk.collect.android-fpP4T_tgz8x_GU7w9kKqvA==/base.apk!/lib/x86_64, /system/lib64, /system/product/lib64]]] couldn't find "libmapbox-maps.so"
        at java.lang.Runtime.loadLibrary0(Runtime.java:1067)
        at java.lang.Runtime.loadLibrary0(Runtime.java:1007)
        at java.lang.System.loadLibrary(System.java:1667)
        at com.mapbox.common.loader.MapboxLibraryLoader.load(MapboxLibraryLoader.kt:19)
        at com.mapbox.maps.loader.MapboxMapsInitializer.create(MapboxMapsInitializer.java:19)
        at com.mapbox.maps.loader.MapboxMapsInitializer.create(MapboxMapsInitializer.java:16)
        at androidx.startup.AppInitializer.doInitialize(AppInitializer.java:180)
        at androidx.startup.AppInitializer.discoverAndInitialize(AppInitializer.java:238) 
        at androidx.startup.AppInitializer.discoverAndInitialize(AppInitializer.java:206) 
        at androidx.startup.InitializationProvider.onCreate(InitializationProvider.java:45) 
        at android.content.ContentProvider.attachInfo(ContentProvider.java:2092) 
        at android.content.ContentProvider.attachInfo(ContentProvider.java:2066) 
        at android.app.ActivityThread.installProvider(ActivityThread.java:6983) 
        at android.app.ActivityThread.installContentProviders(ActivityThread.java:6528) 
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6445) 
        at android.app.ActivityThread.access$1300(ActivityThread.java:219) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1859) 
```

it takes place immediately when the app is started 

Here is the size of those files:
![image](https://user-images.githubusercontent.com/3276264/168651755-ff04ec9e-bb37-429c-8bc7-93e3069f9f06.png)

Reply:
If you haven't already, could you please review https://github.com/getodk/collect/pull/3108 and https://github.com/getodk/collect/pull/3118 to see if there are any ideas there?

The error you shared looks like it would be from a x86_64 device. Was it? I would not expect an arm device to give that error.

Reply:
If this error is indeed only on architectures with missing libs (which it looks like and hopefully is), then that pushes us more towards not including Mapbox in all builds (using product flavours or my preferred conditional dependency method). That would allow us to continue to exclude x86 libs as we'd just use the "non mapbox" build types on emulators.

Reply:
> If this error is indeed only on architectures with missing libs

Yes the crash occurs only on x86 devices (if I remove those libs).